sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -15,19 +20,42 @@ async fn test_get_blob_properties() -> Result<(), Box<dyn Error>> {
     let credential = DefaultAzureCredentialBuilder::default().build()?;
 
     // Act
+    let container_client = BlobContainerClient::new(
+        &endpoint,
+        String::from("testcontainer1"),
+        credential.clone(),
+        None,
+    )?;
+    container_client.create_container(None).await?;
+
     let blob_client = BlobClient::new(
         &endpoint,
-        String::from("testcontainer"),
+        String::from("testcontainer1"),
         String::from("test_blob.txt"),
         credential,
         Some(BlobClientOptions::default()),
     )?;
+    let data = b"hello rusty world";
+    blob_client
+        .upload_blob(
+            RequestContent::from(data.to_vec()),
+            true,
+            data.len() as i64,
+            None,
+        )
+        .await?;
     let response = blob_client
         .get_blob_properties(Some(BlobBlobClientGetPropertiesOptions::default()))
         .await;
 
     // Assert
     assert!(response.is_ok());
+
+    let blob_properties = response?;
+    assert_eq!(blob_properties.blob_type, Some(BlobType::BlockBlob));
+    assert_eq!(blob_properties.content_length, Some(17));
+
+    container_client.delete_container(None).await?;
```

Comment:
nit: Move all of these to a try-finally so cleanup attempt happens even if the test fails unexpectedly.

Reply:
There is no try-catch or try-finally since Rust is a Results-based driven codeflow. I found some workarounds but they seem way too convoluted. Going to punt this for a little longer.